File: vlc-remote-app-flutter/lib/widgets/others/time_indicator.dart
import 'package:flutter/material.dart';

import '../../utils/regex.dart';
import '../../utils/sizer.dart';

class TimeIndicator extends StatelessWidget {
    TimeIndicator({
        @required this.time,
    });

    final int time;

    @override
    Widget build(BuildContext context) {
        Duration duration = Duration(seconds: time);
        return Text(
            duration.toString().replaceFirst(zeroHourRegex, "").split(".")[0],
            style: TextStyle(
                color: Theme.of(context).colorScheme.onBackground,
                fontSize: Sizer.fontSize(9.0),
            ),
        );
    }
}

File: vlc-remote-app/components/remote/TimeIndicator.js
import React from 'react';

import { Text } from 'react-native-paper';

import Duration from '../../models/Duration';

const TimeIndicator = ({ time, style }) => {
    const duration = new Duration(time);

    return (
        <Text style={style}>
            { duration.toString() }
        </Text>
    );
};

export default TimeIndicator;